packages/schema/src/idl.rs

```diff
@@ -0,0 +1,104 @@
+//! The Cosmwasm IDL (Interface Description Language)
+
+use std::collections::BTreeMap;
+
+use schemars::schema::RootSchema;
+use thiserror::Error;
+
+/// The version of the CosmWasm IDL.
+///
+/// Follows Semantic Versioning 2.0.0: https://semver.org/
+///
+/// To determine if a change is breaking, assume consumers allow unknown fields.
+pub const IDL_VERSION: &str = "1.0.0";
+
+/// Rust representation of a contract's API.
+pub struct Api {
+    pub contract_name: String,
+    pub contract_version: String,
+    pub instantiate: RootSchema,
+    pub execute: RootSchema,
+    pub query: RootSchema,
```

Comment:
`instantiate` is the only required import. `execute`/`query` should also be `Option`s.